amxmodx/gameconfigs.cpp

```diff
@@ -0,0 +1,165 @@
+// vim: set ts=4 sw=4 tw=99 noet:
+//
+// AMX Mod X, based on AMX Mod by Aleksander Naszko ("OLO").
+// Copyright (C) The AMX Mod X Development Team.
+//
+// This software is licensed under the GNU General Public License, version 3 or higher.
+// Additional exceptions apply. For full license details, see LICENSE.txt or visit:
+//     https://alliedmods.net/amxmodx-license
+
+#include "gameconfigs.h"
+#include "amxmodx.h"
+#include "CGameConfigs.h"
+
+Handle<GameConfigNative> GameConfigHandle;
+
+// native GameConfig:LoadGameConfigFile(const file[]);
+static cell AMX_NATIVE_CALL LoadGameConfigFile(AMX *amx, cell *params)
+{
+	int length;
+	const char *filename = get_amxstring(amx, params[1], 0, length);
+
+	IGameConfig *config;
+	char error[128];
+
+	if (!ConfigManager.LoadGameConfigFile(filename, &config, error, sizeof(error)))
+	{
+		LogError(amx, AMX_ERR_NATIVE, "Unable to open %s: %s", filename, error);
+		return 0;
+	}
+
+	int handle = GameConfigHandle.create();
+
+	GameConfigHandle.lookup(handle)->m_config = config;
```

Comment:
While this *should* be safe I'd like it better if it properly checked the handle for null.